### PR TITLE
Parsing `<H1>` fails?

### PR DESCRIPTION
**Pull request by fisx**
I feel like this has to be me...

### CODE REVIEW

**Comment by fisx** [comment]
I'll look into this.

**Comment by fisx** [comment]
It was easy enough to fix.  :-)

(Note that travis is still acting up.  See comments in #6.)

File: Text/HTML/Parser.hs [hunk 1]
```diff
@@ -106,7 +106,7 @@ tagOpen =
 -- TODO: This isn't right
 endTagOpen :: Parser Token
 endTagOpen = do
-    name <- takeWhile $ \c -> isAsciiUpper c || isAsciiLower c
+    name <- takeWhile $ notInClass "\x09\x0a\x0c />"
```
Comment (bgamari) [inline Text/HTML/Parser.hs]: Hmm, this is quite odd; at first glance the [specification](http://w3c.github.io/html/syntax.html#end-tag-open-state) seems rather explicit here but the `TODO` above suggests that I approximated here. I'm afraid this is all very much paged out at this point. Do you have any insight into what is happening here?

**Comment by bgamari** [comment]
Seems plausible but I don't quite see how this aligns with the spec.

File: Text/HTML/Parser.hs (hunk 1 above)
Comment (fisx) [reply Text/HTML/Parser.hs]: I don't know anything, it just seemed like the right change and it fixed my problem, and I was hoping you'd confirm that it makes sense.

I think your code neglected to "then switch to the tag name state" (8.2.4.10).  That would mean that I still didn't get it right.  I'll try again.

**Comment by fisx** [comment]
This aligns more with the spec (I think), but I'm not sure I like the spec even a bit.  :-)

My first attempt at fixing this was too lenient, and I didn't require the first tag name character to be alpha.

**Comment by bgamari** [comment]
This looks much better. Also, while reading the spec yesterday I had missed the "anything else" case; I now see that yes, this is indeed compliant with the defined syntax.

**Pull request merged**